### DISCUSSION TITLE
Handle usb-c displays

### DISCUSSION THREAD

**Discussion by Porkepix** [fwupd/fwupd#3674] [Q&A]
**Describe the question**

I have usb-c displays at work, which seems badly detected even though `fwupd` does in fact detect something, and after a quick look at the supported device list, I don't get the feeling there currently are any usb-c displays. Is there any plans for it at some point?

**Comment by hughsie** [comment]
What exact hardware is this?

**Reply by Porkepix** [reply]
The one at work is a Dell P2720DC, but they now offer several others models in their P and U ranges with that.
Displays offer usb-c connectivity which provide laptop charging feature (Power Delivery), video with multiplexed Display Port, and usb hub in a single cable (the display have several usb A ports). Some of them also include ethernet with this.

The point is, those displays have firmware that can be updated at least when connected with the usb-c (to be honest I have no idea how is updated the firmware of displays with hdmi/display port/dvi/vga only, I never took a look but maybe some of those also allow for updates). What Dell provide for this are obviously Windows software. Other manufacturers also start providing Power delivery displays, or sometimes lower versions delivering enough power for a smartphone but not a laptop (can vary from 10 to 100W depending on the implementation).

**Reply by hughsie** [reply]
@dell-client-linux could this be in scope? Lenovo support fwupd for some of the dock hardware now.

**Reply by superm1** [reply]
>which seems badly detected even though fwupd does in fact detect something

Can you share more about what it detects?  A snippet of a verbose daemon log when it's plugged in is useful information.

>to be honest I have no idea how is updated the firmware of displays with hdmi/display port/dvi/vga only, I never took a look but maybe some of those also allow for updates

Some of these firmwares can be updated via DP-aux, but others might be an out of band ISP update only.

>@dell-client-linux could this be in scope? Lenovo support fwupd for some of the dock hardware now.

Thus far Dell supports docks only, but you certainly can make an argument that these are monitors with many of the same components of docks inside of them.
